pkg/registration/activate.go

```diff
@@ -43,7 +43,7 @@ type activateResponse struct {
 // Activate a product by pairing an authorized connection (which contains the
 // system at hand), plus the "triplet" being used to identify the desired
 // product.
-func Activate(conn connection.Connection, identifier, version, arch, regcode string) (*Metadata, *Product, error) {
+func Activate(conn connection.Connection, identifier, version, arch, regcode string) (*Metadata, Product, error) {
```

Comment:
Why removing the point to product here? Is there a specific case why this would be desirable? I'm wondering why keeping `*Metadata` as pointer but not `Product`?